packages/sources/coinmetrics-test/src/endpoint/total-burned.ts

```diff
@@ -0,0 +1,248 @@
+    type: 'string',
+    required: true,
+  },
+  frequency: {
+    description: 'At which interval to calculate the number of coins/tokens burned',
+    type: 'string',
+    required: false,
+    options: FrequencyInputOptions,
+    default: Frequency.ONE_DAY,
+  },
+  pageSize: {
+    description: 'Number of results to get per page. From 1 to 10000',
+    default: 10_000,
+    type: 'number',
+    required: false,
+  },
+  startTime: {
+    description:
+      'The start time for the queried period. See [Supported DateTime Formats](#supported-datetime-formats)',
+    type: 'string',
+    required: false,
+  },
+  endTime: {
+    description:
+      'The end time for the queried period. See [Supported DateTime Formats](#supported-datetime-formats)',
+    type: 'string',
+    required: false,
+  },
+} as const
+
+export const calculateBurnedTKN = (assetMetricsList: AssetMetrics[]): BigNumber => {
+  let burnedTKN = BigNumber.from('0')
+  assetMetricsList.forEach((assetMetrics: AssetMetrics) => {
+    let feeTotNTV
+    let revNtv
+    let issTotNtv
+    try {
+      feeTotNTV = ethers.utils.parseEther(assetMetrics.FeeTotNtv)
+      revNtv = ethers.utils.parseEther(assetMetrics.RevNtv)
+      issTotNtv = ethers.utils.parseEther(assetMetrics.IssTotNtv)
+    } catch (error) {
+      throw new AdapterError({
+        message: `Unprocessable asset metrics: ${JSON.stringify(assetMetrics)}, due to: ${error}.`,
+      })
+    }
+    burnedTKN = burnedTKN.add(feeTotNTV.sub(revNtv.sub(issTotNtv)))
+  })
+  return burnedTKN
+}
+
+export class TotalBurnedTransport implements Transport<EndpointTypes> {
+  name!: string
+  requester!: Requester
+  responseCache!: ResponseCache<{
+    Request: EndpointTypes['Request']
+    Response: EndpointTypes['Response']
+  }>
+
+  async initialize(
+    dependencies: TransportDependencies<EndpointTypes>,
+    _adapterSettings: EndpointTypes['Settings'],
+    _endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    this.responseCache = dependencies.responseCache
+    this.requester = dependencies.requester
+    this.name = transportName
+  }
+
+  async foregroundExecute(
+    req: AdapterRequest<EndpointTypes['Request']>,
+    settings: typeof config.settings,
+  ): Promise<AdapterResponse<EndpointTypes['Response']>> {
+    let totalBurnedTKN = BigNumber.from('0')
+
+    let lastPage = false
+    const input = req.requestContext.data
+    const isBurnedEndpoint = req.requestContext.endpointName === 'burned'
+    if (isBurnedEndpoint) {
+      input.pageSize = 1
+    }
+    const requestConfig = this.prepareRequest(req.requestContext.data, settings)
+
+    const providerDataRequestedUnixMs = Date.now()
+    while (!lastPage) {
+      const result = await this.requester.request<ResponseSchema>(
+        calculateHttpRequestKey({
+          context: {
+            adapterSettings: settings,
+            inputParameters: inputParams,
+            endpointName: ENPDOINT_NAME,
```

Comment:
Shouldn't this be set to the endpoint name in the request?
```suggestion
            endpointName: req.requestContext.endpointName,
```